### PR TITLE
'$' EoL Motion

### PR DESCRIPTION
**Pull request by josephliccini**
Hey guys

Wrote the $ motion, but I wasn't sure where to put logic for registering a command that isn't in VSCode Built in.

Also, I what I've written isn't particularly reusable for combining with another motion (i.e. 'd$', or 'y$')..

Feedback / thoughts on where to put the 'cursorEndOfLine' command?

Thanks
Joe

### CODE REVIEW

**Comment by jpoon** [comment]
Thanks @josephliccini!  As we don't really have a visual mode yet, would you mind adding this to a new `$` handler to `/src/mode/modeNormal.ts`? 

We need to come up with a better strategy of combining motions but current focus is to get a quick initial release out the door.

**Comment by jpoon** [comment]
(Argh. Accidentally pressed the `x` button to delete my previous message).

Thanks @josephliccini! As we haven't put any effort in the visual mode yet, would you mind putting your `cursorEndOfLine` logic to the `$` delegate that is in `src/mode/modeNormal.ts`? 

We don't yet have a good strategy for combining with other motions; want to push out a small release soon!

File: extension.ts [hunk 1]
```diff
@@ -96,9 +96,32 @@ export function activate(context: vscode.ExtensionContext) {
...
+    
+    registerCustomCommands();
     context.subscriptions.push(cmdLineDisposable);
 }
 
 function handleKeyEvent(key:string) {
     modeHandler.handleKeyEvent(key);
 }
+
+function registerCustomCommands() {
```
Comment (jpoon) [inline extension.ts]: Move this inside `"$": () => { <Here> }`? instead of `modeNormal.ts` instead?

File: extension.ts [hunk 2]
```diff
@@ -96,9 +96,32 @@ export function activate(context: vscode.ExtensionContext) {
...
+        const lineLength = vscode.window.activeTextEditor.document.lineAt(lineNumber).text.length;
+        const charDelta = lineLength - cursorCharIndex;
+        
+        const position = new vscode.Position(lineNumber, cursorCharIndex);
+        const updatedPosition = position.translate(0, charDelta);
+        
+        const selection = new vscode.Selection(updatedPosition, updatedPosition);
+        
+        vscode.window.activeTextEditor.selection = selection;
+        vscode.window.activeTextEditor.revealRange(selection, vscode.TextEditorRevealType.Default);
```
Comment (jpoon) [inline extension.ts]: I haven't tested this out but would this be equivalent to one of the built-in VS commands: `cursorEnd` (e.g. `vscode.commands.executeCommand("cursorEnd")`?

**Comment by josephliccini** [comment]
Came up with a way that matches what you guys have put together with the Cursor class coming in soon

**Pull request closed without merging**